docs/communicate-with-javascript.md

```diff
@@ -63,31 +65,60 @@ var a = 1; var b = 2
 var add = a + b
 ```
 
-The difference between one and two `%` characters is detailed in the [OCaml
-documentation](https://v2.ocaml.org/manual/extensionnodes.html).
+The difference between one and two percentage characters is detailed in the
+[OCaml documentation](https://v2.ocaml.org/manual/extensionnodes.html).
+
+#### Attributes
 
 Attributes are "decorations" applied to specific parts of the code to provide
-additional information. Melange uses attributes in various ways to enhance
-communication with JavaScript code. For instance, it introduces the `bs.as`
-attribute, which allows renaming of fields in a record on the generated
-JavaScript code:
+additional information. In Melange, attributes are used in two ways to enhance
+the expressiveness of generating JavaScript code.
 
-```ocaml
-type t = {
-  foo : int; [@bs.as "foo_for_js"]
-  bar : string;
-}
+The first approach is leveraging the existing [OCaml's built-in
```

Comment:
you describe the "what" but not the "why". I think we should at least include a tiny sentence that says it's useful for heterogeneous collections or something.

Reply:
added in https://github.com/melange-re/documentation-site/pull/35/commits/d9d6c637fc410167d553d33652e3e84b223d3409.